Cannot be built with GHC 9.2.0.20210821

```
src/Data/Serialize/Get.hs:726:36: error:
    • Couldn't match expected type ‘Word16#’ with actual type ‘Word#’
    • In the first argument of ‘W16#’, namely
        ‘(w `uncheckedShiftL#` i)’
      In the expression: W16# (w `uncheckedShiftL#` i)
      In an equation for ‘shiftl_w16’:
          shiftl_w16 (W16# w) (I# i) = W16# (w `uncheckedShiftL#` i)
    |
726 | shiftl_w16 (W16# w) (I# i) = W16# (w `uncheckedShiftL#`   i)
    |                                    ^^^^^^^^^^^^^^^^^^^^^^^^

src/Data/Serialize/Get.hs:727:36: error:
    • Couldn't match expected type ‘Word32#’ with actual type ‘Word#’
    • In the first argument of ‘W32#’, namely
        ‘(w `uncheckedShiftL#` i)’
      In the expression: W32# (w `uncheckedShiftL#` i)
      In an equation for ‘shiftl_w32’:
          shiftl_w32 (W32# w) (I# i) = W32# (w `uncheckedShiftL#` i)
    |
727 | shiftl_w32 (W32# w) (I# i) = W32# (w `uncheckedShiftL#`   i)
    |                                    ^^^^^^^^^^^^^^^^^^^^^^^^
```

--- a/src/Data/Serialize/Get.hs
+++ b/src/Data/Serialize/Get.hs
@@ -723,8 +723,13 @@ shiftl_w32 :: Word32 -> Int -> Word32
 shiftl_w64 :: Word64 -> Int -> Word64
 
 #if defined(__GLASGOW_HASKELL__) && !defined(__HADDOCK__)
+#if MIN_VERSION_base(4,16,0)
+shiftl_w16 (W16# w) (I# i) = W16# (w `uncheckedShiftLWord16#` i)
+shiftl_w32 (W32# w) (I# i) = W32# (w `uncheckedShiftLWord32#` i)
+#else
 shiftl_w16 (W16# w) (I# i) = W16# (w `uncheckedShiftL#`   i)
 shiftl_w32 (W32# w) (I# i) = W32# (w `uncheckedShiftL#`   i)
+#endif
 
 #if WORD_SIZE_IN_BITS < 64
 shiftl_w64 (W64# w) (I# i) = W64# (w `uncheckedShiftL64#` i)

